dids/src/main/kotlin/web5/sdk/dids/methods/dht/Dht.kt

```diff
@@ -0,0 +1,277 @@
+     * @param seq The sequence number of the message.
+     * @param v The value to be written to the DHT.
+     * @return A BEP44 signed message [Bep44Message].
+     * @throws IllegalArgumentException if the private key is not an Ed25519 key.
+     * @throws IllegalArgumentException if the value to sign is empty.
+     * @throws IllegalArgumentException if the length of the compressed v value is > 1000 bytes.
+     */
+    public fun signBep44Message(manager: KeyManager, keyAlias: String, seq: Long, v: ByteArray): Bep44Message {
+      // get the public key to verify it is an Ed25519 key
+      val pubKey = manager.getPublicKey(keyAlias)
+      require(
+        pubKey.keyType == Ed25519.keyType &&
+          pubKey.algorithm == Ed25519.algorithm
+      ) {
+        "Must supply an Ed25519 key"
+      }
+
+      // encode v using bencode
+      val out = ByteArrayOutputStream()
+      BEncoder.bencode(v, out)
+      val vEncoded = out.toByteArray()
+
+      require(vEncoded.size <= 1000) {
+        "Value must be <= 1000 bytes compressed, current bytes {${vEncoded.size}}"
+      }
+
+      // encode according to BEP44
+      val bytesToSign = "3:seqi${seq}e1:v".toByteArray() + vEncoded
+
+      // sign and return the BEP44 message
+      manager.sign(keyAlias, bytesToSign).let { signature ->
+        return Bep44Message(
+          v = v,
+          k = Ed25519.publicKeyToBytes(pubKey),
+          sig = signature,
+          seq = seq
+        )
+      }
+    }
+
+    /**
+     * Verifies a message according to the BEP44 Signature Verification specification.
+     * https://www.bittorrent.org/beps/bep_0044.html
+     *
+     * @param message The message to verify.
+     * @return True if the message is verified, false otherwise.
+     * @throws IllegalArgumentException if the Bep44Message is malformed.
+     * @throws SignatureException if the signature is invalid.
+     */
+    public fun verifyBep44Message(message: Bep44Message) {
+      // encode v using bencode
+      val out = ByteArrayOutputStream()
+      BEncoder.bencode(message.v, out)
+      val vEncoded = out.toByteArray()
+
+      // prepare buffer and verify
+      val bytesToVerify = "3:seqi${message.seq}e1:v".toByteArray() + vEncoded
+
+      // create a JWK representation of the public key
+      val ed25519PublicKey = Ed25519.bytesToPublicKey(message.k)
+
+      // verify the signature
+      Ed25519.verify(ed25519PublicKey, bytesToVerify, message.sig)
+    }
+  }
+}
+
+/**
+ * Complies with BEP-44 https://www.bittorrent.org/beps/bep_0044.html
+ * Implements a subset of BEP-44 for Mutable Items that omits salt and CAS values.
+ *
+ * @property v The value to publish.
+ * @property k The 32 byte representation of an Ed25519 key to publish the value under.
+ * @property sig The 64 byte EdDSA signature over the value and sequence number.
+ * @property seq The sequence number of the message.
+ */
+public data class Bep44Message(
```

Comment:
its a data class because its purpose is to hold data..it does not need what a regular class offers. unless I am misunderstanding the purpose of the data class. it's like a POJO

Reply:
it's public because the (now renamed) `DhtClient` is public and needed for testing. that's another benefit of making this public...more test coverage.

Reply:
updated to internal for testing